functions/go/bind/pkg/meta/namespaceref.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package meta
+
+import (
+	"fmt"
+
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+type namespaceRef struct {
+	parentObject *fn.SubObject
+
+	name      string
+	fieldPath []string
+}
+
+func buildMetadataNamespaceReference(parentObject *fn.KubeObject) (Ref, error) {
+	namespace := parentObject.GetMap("metadata").GetString("namespace")
```

Comment:
`parentObject.GetNamespace()` gives the same result I think. the `GetMap` can return `nil` and then causes panic I think https://github.com/GoogleContainerTools/kpt-functions-sdk/blob/master/go/fn/object.go#L962